src/pipecat/services/speechmatics/stt.py

```diff
@@ -0,0 +1,608 @@
+    InterimTranscriptionFrame,
+    StartFrame,
+    TranscriptionFrame,
+)
+from pipecat.services.stt_service import STTService
+from pipecat.transcriptions.language import Language
+from pipecat.utils.time import time_now_iso8601
+
+
+class AudioBuffer:
+    """Audio buffer for STT clients."""
+
+    def __init__(self, maxsize: int = 0):
+        self._queue = asyncio.Queue(maxsize=maxsize)
+        self._current_chunk = b""
+        self._position = 0
+        self._closed = False
+
+    def write_audio(self, data: bytes) -> None:
+        """Write audio data to the buffer (thread-safe)."""
+        if data:
+            try:
+                self._queue.put_nowait(data)
+            except asyncio.QueueFull:
+                pass
+
+    async def read(self, size: int) -> bytes:
+        """Read exactly `size` bytes from the buffer."""
+        result = b""
+        bytes_needed = size
+
+        while bytes_needed > 0 and not self._closed:
+            # Use data from current chunk if available
+            if self._position < len(self._current_chunk):
+                available = len(self._current_chunk) - self._position
+                take = min(bytes_needed, available)
+                result += self._current_chunk[self._position : self._position + take]
+                self._position += take
+                bytes_needed -= take
+                continue
+
+            # Get next chunk
+            try:
+                chunk = await asyncio.wait_for(self._queue.get(), timeout=0.1)
+                if chunk is None:
+                    continue
+                self._current_chunk = chunk
+                self._position = 0
+            except asyncio.TimeoutError:
+                await asyncio.sleep(0)
+                continue
+
+        return result
+
+    def stop(self) -> None:
+        """Close the audio buffer."""
+        self._closed = True
+
+
+@dataclass
+class SpeechFragment:
+    """Fragment of an utterance.
+
+    Attributes:
+        start_time: Start time of the fragment in seconds (from session start).
+        end_time: End time of the fragment in seconds (from session start).
+        language: Language of the fragment.
+            Defaults to `en`.
+        is_eos: Whether the fragment is the end of a sentence.
+            Defaults to `False`.
+        is_final: Whether the fragment is the final fragment.
+            Defaults to `False`.
+        attaches_to: Whether the fragment attaches to the previous or next fragment (punctuation).
+            Defaults to empty string.
+        content: Content of the fragment.
+            Defaults to empty string.
+        speaker: Speaker of the fragment (if diarization is enabled).
+            Defaults to `None`.
+        confidence: Confidence of the fragment (0.0 to 1.0).
+            Defaults to `1.0`.
+        result: Raw result of the fragment.
+            Defaults to `None`.
+    """
+
+    start_time: float
+    end_time: float
+    language: str = "en"
```

Comment:
Let's use Pipecat's Language module in place of strings.

```suggestion
    language: str = Language.EN
```